src/Assert.php

```diff
@@ -1602,6 +1604,24 @@ public static function keyNotExists($array, $key, $message = '')
         }
     }
 
+    /**
+     * Checks if a value is a valid array key (int or string).
+     *
+     * @param string|int $key
```

Comment:
This should be mixed, its the assertions job to make sure this is a string or int